draft-ietf-quic-qcram.md

```diff
@@ -223,15 +331,111 @@ It is an error if the HPACK decoder encounters an indexed representation that
+using the index of that entry. In this case, the `S` bit indicates whether the
+reference is to the static (S=1) or dynamic (S=0) table and the index of the
+entry is represented as an integer with an 5-bit prefix (see Section 5.1 of
+[RFC7541]). Valid table indices are always non-zero; a table index of zero
+MUST be treated as a decoding error.
 
-Due to out-of-order arrival, QCRAM's eviction algorithm requires changes
-(relative to HPACK) to avoid the possibility that an indexed representation is
-decoded after the referenced entry has already been evicted.  QCRAM employs a
-two-phase eviction algorithm, in which the encoder will not evict entries that
-have outstanding (unacknowledged) references.
+~~~~~~~~~~
+     0   1   2   3   4   5   6   7
+   +---+---+---+---+---+---+---+---+
+   | 0 | N | S |  Name Index (5+)  |
+   +---+---+-----------------------+
+   | H |     Value Length (7+)     |
+   +---+---------------------------+
+   | Value String (Length octets)  |
+   +-------------------------------+
+~~~~~~~~~~
+{: title="Literal Header Field -- Indexed Name"}
 
-### Blocked Evictions
+Otherwise, the header field name is represented as a string literal (see Section
+5.2 of [RFC7541]). A value 0 is used in place of the 6-bit index, followed by
+the header field name.
+
+~~~~~~~~~~
+     0   1   2   3   4   5   6   7
+   +---+---+---+---+---+---+---+---+
+   | 0 | N |           0           |
+   +---+---+-----------------------+
+   | H |     Name Length (7+)      |
+   +---+---------------------------+
+   |  Name String (Length octets)  |
+   +---+---------------------------+
+   | H |     Value Length (7+)     |
+   +---+---------------------------+
+   | Value String (Length octets)  |
+   +-------------------------------+
+~~~~~~~~~~
+{: title="Literal Header Field -- Literal Name"}
+
+Either form of header field name representation is followed by the header field
+value represented as a string literal (see Section 5.2 of [RFC7541]).
+
+
+# Encoding Strategies
+
+## Reference Tracking
+
+An encoder MUST ensure that a header block which references a dynamic table
+entry is not received by the decoder after the referenced entry has already been
+evicted, and might wish to ensure that the decoder will not suffer head-of-line
+blocking when encoding particular references.
+
+In order to enable this, the encoder MUST track outstanding (unacknowledged)
+header blocks and MAY track outstanding table updates.
+
+When the encoder receives feedback from the decoder, it dereferences table
+entries that were indexed in the acknowledged header.  To track which entries
+must be dereferenced, it can maintain a map from unacknowledged headers to lists
+of (absolute) indices.  The simplest place to store the actual reference count
+might be the table entries.  In practice the number of entries in the table with
+a non-zero reference count is likely to stay quite small.  A data structure
```

Comment:
You don't actually need to maintain a reference count, it's only one potential way of doing this (admittedly, a pretty reasonable one).  I'd prefer not to reference (or assume) a particular implementation strategy here.